rows_picker.go

```diff
@@ -0,0 +1,1018 @@
+		}{
+			r,
+			wrappedRowsColumnTypeDatabaseTypeName{r.parent},
+			wrappedRowsColumnTypeNullable{r.parent},
+			wrappedRowsColumnTypePrecisionScale{r.parent},
+			wrappedRowsColumnTypeScanType{r.parent},
+		}
+	}
+
+	// plain driver.Rows
+	pickRows[59] = func(r *wrappedRows) driver.Rows {
+		return struct {
+			*wrappedRows
+			wrappedRowsNextResultSet
+			wrappedRowsColumnTypeDatabaseTypeName
+			wrappedRowsColumnTypeNullable
+			wrappedRowsColumnTypePrecisionScale
+			wrappedRowsColumnTypeScanType
+		}{
+			r,
+			wrappedRowsNextResultSet{r.parent},
+			wrappedRowsColumnTypeDatabaseTypeName{r.parent},
+			wrappedRowsColumnTypeNullable{r.parent},
+			wrappedRowsColumnTypePrecisionScale{r.parent},
+			wrappedRowsColumnTypeScanType{r.parent},
+		}
+	}
+
+	// plain driver.Rows
+	pickRows[60] = func(r *wrappedRows) driver.Rows {
+		return struct {
+			*wrappedRows
+			wrappedRowsColumnTypeLength
+			wrappedRowsColumnTypeNullable
+			wrappedRowsColumnTypePrecisionScale
+			wrappedRowsColumnTypeScanType
+		}{
+			r,
+			wrappedRowsColumnTypeLength{r.parent},
+			wrappedRowsColumnTypeNullable{r.parent},
+			wrappedRowsColumnTypePrecisionScale{r.parent},
+			wrappedRowsColumnTypeScanType{r.parent},
+		}
+	}
+
+	// plain driver.Rows
+	pickRows[61] = func(r *wrappedRows) driver.Rows {
+		return struct {
+			*wrappedRows
+			wrappedRowsNextResultSet
+			wrappedRowsColumnTypeLength
+			wrappedRowsColumnTypeNullable
+			wrappedRowsColumnTypePrecisionScale
+			wrappedRowsColumnTypeScanType
+		}{
+			r,
+			wrappedRowsNextResultSet{r.parent},
+			wrappedRowsColumnTypeLength{r.parent},
+			wrappedRowsColumnTypeNullable{r.parent},
+			wrappedRowsColumnTypePrecisionScale{r.parent},
+			wrappedRowsColumnTypeScanType{r.parent},
+		}
+	}
+
+	// plain driver.Rows
+	pickRows[62] = func(r *wrappedRows) driver.Rows {
+		return struct {
+			*wrappedRows
+			wrappedRowsColumnTypeDatabaseTypeName
+			wrappedRowsColumnTypeLength
+			wrappedRowsColumnTypeNullable
+			wrappedRowsColumnTypePrecisionScale
+			wrappedRowsColumnTypeScanType
+		}{
+			r,
+			wrappedRowsColumnTypeDatabaseTypeName{r.parent},
+			wrappedRowsColumnTypeLength{r.parent},
+			wrappedRowsColumnTypeNullable{r.parent},
+			wrappedRowsColumnTypePrecisionScale{r.parent},
+			wrappedRowsColumnTypeScanType{r.parent},
+		}
+	}
+
+	// plain driver.Rows
+	pickRows[63] = func(r *wrappedRows) driver.Rows {
+		return struct {
+			*wrappedRows
+			wrappedRowsNextResultSet
+			wrappedRowsColumnTypeDatabaseTypeName
+			wrappedRowsColumnTypeLength
+			wrappedRowsColumnTypeNullable
+			wrappedRowsColumnTypePrecisionScale
+			wrappedRowsColumnTypeScanType
+		}{
+			r,
+			wrappedRowsNextResultSet{r.parent},
+			wrappedRowsColumnTypeDatabaseTypeName{r.parent},
+			wrappedRowsColumnTypeLength{r.parent},
+			wrappedRowsColumnTypeNullable{r.parent},
+			wrappedRowsColumnTypePrecisionScale{r.parent},
+			wrappedRowsColumnTypeScanType{r.parent},
+		}
+	}
+}
+
+type RowsUnwrapper interface {
+	RowsUnwrap() driver.Rows
+}
```

Comment:
this can be defined in a non-generated file. it also needs to be documented for users of the library